Build Error on the latest gradle: Gradle-6.5.+

When building the app using the latest Gradle version (6.5.+) the following exception is occurred: 

`Note: Recompile with -Xlint: deprecation for details.
e: C:\Users\Melak\AppData\Local\Pub\Cache\hosted\pub.dartlang.org\in_app_update-1.1.14\android\src\main\kotlin\de\ffuf\in_app_update\InAppUpdatePlugin.kt: (30, 1): Class 'InAppUpdatePlugin' is not abstract and does not implement abstract member public abstract fun onActivityCreated(p0: Activity, p1: Bundle?): Unit defined in android.app.Application.ActivityLifecycleCallbacks
e: C:\Users\Melak\AppData\Local\Pub\Cache\hosted\pub.dartlang.org\in_app_update-1.1.14\android\src\main\kotlin\de\ffuf\in_app_update\InAppUpdatePlugin.kt: (114, 5): 'onActivityCreated' overrides nothing
e: C:\Users\Melak\AppData\Local\Pub\Cache\hosted\pub.dartlang.org\in_app_update-1.1.14\android\src\main\kotlin\de\ffuf\in_app_update\InAppUpdatePlugin.kt: (116, 5): 'onActivityPaused' overrides nothing
e: C:\Users\Melak\AppData\Local\Pub\Cache\hosted\pub.dartlang.org\in_app_update-1.1.14\android\src\main\kotlin\de\ffuf\in_app_update\InAppUpdatePlugin.kt: (118, 5): 'onActivityStarted' overrides nothing
e: C:\Users\Melak\AppData\Local\Pub\Cache\hosted\pub.dartlang.org\in_app_update-1.1.14\android\src\main\kotlin\de\ffuf\in_app_update\InAppUpdatePlugin.kt: (120, 5): 'onActivityDestroyed' overrides nothing
e: C:\Users\Melak\AppData\Local\Pub\Cache\hosted\pub.dartlang.org\in_app_update-1.1.14\android\src\main\kotlin\de\ffuf\in_app_update\InAppUpdatePlugin.kt: (122, 5): 'onActivitySaveInstanceState' overrides nothing
e: C:\Users\Melak\AppData\Local\Pub\Cache\hosted\pub.dartlang.org\in_app_update-1.1.14\android\src\main\kotlin\de\ffuf\in_app_update\InAppUpdatePlugin.kt: (124, 5): 'onActivityStopped' overrides nothing
e: C:\Users\Melak\AppData\Local\Pub\Cache\hosted\pub.dartlang.org\in_app_update-1.1.14\android\src\main\kotlin\de\ffuf\in_app_update\InAppUpdatePlugin.kt: (126, 5): 'onActivityResumed' overrides nothing

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':in_app_update:compileDebugKotlin'.`

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,3 +1,6 @@
+## 1.1.15
+* fix android Gradle version (6.5.+) build error (#55). thanks for https://github.com/mig35
+
 ## 1.1.14
 * fix compilation error on iOS (#50)
 

--- a/android/build.gradle
+++ b/android/build.gradle
@@ -2,7 +2,7 @@ group 'de.ffuf.in_app_update'
 version '1.0-SNAPSHOT'
 
 buildscript {
-    ext.kotlin_version = '1.3.50'
+    ext.kotlin_version = '1.4.21'
     repositories {
         google()
         jcenter()

--- a/android/src/main/kotlin/de/ffuf/in_app_update/InAppUpdatePlugin.kt
+++ b/android/src/main/kotlin/de/ffuf/in_app_update/InAppUpdatePlugin.kt
@@ -119,7 +119,7 @@ class InAppUpdatePlugin : FlutterPlugin, MethodCallHandler,
 
     override fun onActivityDestroyed(activity: Activity) {}
 
-    override fun onActivitySaveInstanceState(activity: Activity, outState: Bundle?) {}
+    override fun onActivitySaveInstanceState(activity: Activity, outState: Bundle) {}
 
     override fun onActivityStopped(activity: Activity) {}
 

--- a/android/src/main/kotlin/de/ffuf/in_app_update/InAppUpdatePlugin.kt
+++ b/android/src/main/kotlin/de/ffuf/in_app_update/InAppUpdatePlugin.kt
@@ -111,19 +111,19 @@ class InAppUpdatePlugin : FlutterPlugin, MethodCallHandler,
         activityProvider = null
     }
 
-    override fun onActivityCreated(activity: Activity?, savedInstanceState: Bundle?) {}
+    override fun onActivityCreated(activity: Activity, savedInstanceState: Bundle?) {}
 
-    override fun onActivityPaused(activity: Activity?) {}
+    override fun onActivityPaused(activity: Activity) {}
 
-    override fun onActivityStarted(activity: Activity?) {}
+    override fun onActivityStarted(activity: Activity) {}
 
-    override fun onActivityDestroyed(activity: Activity?) {}
+    override fun onActivityDestroyed(activity: Activity) {}
 
-    override fun onActivitySaveInstanceState(activity: Activity?, outState: Bundle?) {}
+    override fun onActivitySaveInstanceState(activity: Activity, outState: Bundle?) {}
 
-    override fun onActivityStopped(activity: Activity?) {}
+    override fun onActivityStopped(activity: Activity) {}
 
-    override fun onActivityResumed(activity: Activity?) {
+    override fun onActivityResumed(activity: Activity) {
         appUpdateManager
             ?.appUpdateInfo
             ?.addOnSuccessListener { appUpdateInfo ->
@@ -131,18 +131,10 @@ class InAppUpdatePlugin : FlutterPlugin, MethodCallHandler,
                     == UpdateAvailability.DEVELOPER_TRIGGERED_UPDATE_IN_PROGRESS
                     && appUpdateType == AppUpdateType.IMMEDIATE
                 ) {
-                    val foregroundActivity = activity ?: let {
-                        updateResult?.error(
-                          "in_app_update requires an activity",
-                          null,
-                          null
-                        )
-                        return@addOnSuccessListener
-                    }
                     appUpdateManager?.startUpdateFlowForResult(
                       appUpdateInfo,
                       AppUpdateType.IMMEDIATE,
-                      foregroundActivity,
+                      activity,
                       REQUEST_CODE_START_UPDATE
                     )
                 }

--- a/example/android/build.gradle
+++ b/example/android/build.gradle
@@ -6,7 +6,7 @@ buildscript {
     }
 
     dependencies {
-        classpath 'com.android.tools.build:gradle:3.5.0'
+        classpath 'com.android.tools.build:gradle:4.1.1'
         classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"
     }
 }

--- a/example/android/gradle/wrapper/gradle-wrapper.properties
+++ b/example/android/gradle/wrapper/gradle-wrapper.properties
@@ -1,6 +1,6 @@
-#Fri Jun 23 08:50:38 CEST 2017
+#Thu Jan 07 19:37:12 NOVT 2021
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-5.6.2-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-6.5-bin.zip

--- a/pubspec.yaml
+++ b/pubspec.yaml
@@ -1,6 +1,6 @@
 name: in_app_update
 description: Enables In App Updates on Android using the official Android APIs.
-version: 1.1.14
+version: 1.1.15
 author: jonas.bark@ffuf.de
 homepage: https://ffuf.de
 repository: https://github.com/feilfeilundfeil/flutter_in_app_update

